dali/dfu/dfuwu.cpp

```diff
@@ -990,7 +992,14 @@ class CDFUfileSpec: public CLinkedDFUWUchild, implements IDFUfileSpec
                 userdesc->set(username.str(),password.str());
                 Owned<IDistributedFile> file = queryDistributedFileDirectory().lookup(lfn,userdesc,AccessMode::tbdRead,false,false,nullptr,defaultPrivilegedUser);
                 if (file)
-                    return file->getFileDescriptor();
+                {
+                    if (planeName.isEmpty())
+                        return file->getFileDescriptor();
+
+                    Owned<IFileDescriptor> ret = file->getFileDescriptor();
+                    ret->setPlaneName(planeName.str());
```

Comment:
see comment in dafdesc, we should not be adding new methods to IFileDescriptor.
The IFileDescriptor being created here, should contain correct IClusterInfo and group names (that correspond to the dropzone), if they do not, then that's the area that needs attention.

Reply:
In a bare metal environment, I tested 2 IFileDescriptor calls. The getClusterGroupName(0, planeName, nullptr) returns the host of the dropzone, not a dropzone name. The iClusterInfo->queryGroupName() returns empty (iClusterInfo =fdesc->queryClusterNum(0)).

Reply:
I tested them inside the CDistributedFileDirectory::getFDescPermissions(IFileDescriptor *fdesc, ...). The fdesc is from (dfurun.cpp):
    Owned<IFileDescriptor> fdesc = destination->getFileDescriptor(iskey);
    checkPhysicalFilePermissions(fdesc,userdesc,true);
The destination is from IDFUfileSpec *destination = wu->queryUpdateDestination();
All of those are in dfuserver.

Reply:
When despraying a file, the IDFUfileSpec.setSingleFilename(RemoteFilename &rfn) is called for the destination. The rfn has the host. I haven't found where the host is converted to a dropzone name.

Reply:
One possible way to set the dropzone name is in IDFUfileSpec.setMultiFilename(RemoteMultiFilename &rmfn) (Adding an @dropzone attribute in DFU WU?). Before ESP calls setMultiFilename(), the host  ('localhost' in my DD) is set to a SocketEndpoint. Inside the setMultiFilename(), the URL returned by SocketEndpoint.getUrlStr(url) becomes ('127.0.0.1'). The findDropZonePlane() will fail unless we change the findDropZonePlane() to set the '127.0.0.1' to nullptr. @jakesmith please advise.